Run test suite in GH Actions

harlantwood opened the pull request:
- [x] delete old circle and travis stuff

Runs in Ubuntu and Mac.  Would run in Windows too, but I don't think nix works there.

I also came across these lines in circle and travis configs (circle has not run for 2 years, travis I'm not sure) -- maybe these are useful ways to run the tests?  @pospi LMK if you'd like different test commands, or if you want test in GH actions at all.

```
nix-shell --run 'npm i -g pnpm' && nix-shell --run 'npm run dht:sim2h &'
nix-shell --run 'pnpm install --network-concurrency 1 && npm run build && npm run test:integration:test'
```

```
nix-shell --run 'pnpm install && npm run build && npm run test:integration:test'
```

pospi committed:
remove RevisionHash and change all record helper methods to accept raw HeaderHash in preparation for indexing logic changes

DnaAddressable now must be directly storable as an Entry in order for fully-qualified IDs to be encoded in the DHT. This means conversion traits to/from hdk::prelude::Entry must be created, and the inner type of DnaAddressable cannot be ambiguous since this would make implementing such conversions impossible. This makes sense and is a good thing, because it enforces that only tuples of (DnaHash, EntryHash) can be used as identifiers. Implementors should not be linking to revisions but rather records.

harlantwood committed:
Run `cargo test` in GH Actions

harlantwood commented:
Update: looks like travis last ran 2 years ago as well as Circle.  Links:

https://app.circleci.com/pipelines/github/holo-rea/holo-rea
https://travis-ci.org/github/holo-rea/holo-rea

pospi committed:
change calculate_identity_address() to use hash of externally-facing fully qualified address

necessary changes to related trait bounds to support this

pospi commented:
@Connoropolous actually just submitted a PR for github actions, and those are enabled now to generate builds.

Whether we want the test suite to run as part of that process is another matter, possibly an evolving one. At present still maybe 40-50% of the test suite is failing, mostly just because they haven't yet been updated for RSM (see #174). So that's why I haven't yet re-enabled Travis.

There might be a worthwhile interim step where we run tests on one CI system and builds on another. Tests should be run for any change on `sprout`, but builds only get run if you push a version tag. It's possibly also worthwhile keeping them as parallel things on different infrastructure because it means, well, parallelisation.

Eventually though yes I think it's best if everything gets run in GH actions and the build doesn't get kicked off if the tests fail!

Oh, and a note on the test suite- I didn't have many Rust tests in the beginning so was only running Tryorama tests in JS. Eventually I think it would be best to simply run `npm run test`, which first runs unit tests with `cargo test` before triggering the integration test suite in the `test/` subdirectory. Rust tests probably need a QA and some polish before doing that, and the stuff in `test/core-architecture` should be being done with cargo rather than nodejs & tape (see #43).

Does that help? Opinions & input very welcome (:

pospi committed:
remove RevisionHash from remaining core libs & definition

pospi committed:
drop all refs to deprecated RevisionHash in favour of raw HeaderHash

cleanup some remaining unneeded non-generic deletion payload structs

remove unneeded type generics on deletion functions

pospi committed:
fix generics to match new method signatures, missing ByHeader imports in RPC crates

pospi committed:
reimplement identity read/write helpers to directly store UUIDs in the DHT

remove unneeded direct retrieval of identity EntryHash, now gets read from DnaAddressable

pospi committed:
adjust trait bounds for identity managing methods to reflect new storage requirements

pospi committed:
define identifier EntryDefs using verbose macro as substitution for proc macro attributes in macros-by-example does not work

pospi committed:
fix EntryDefs for index pointers not being defined, and use of entry_defs overriding already-registered defs in indexing zomes

pospi committed:
reimplement temporary 'all' index logic to use non-Path-based indexing structure

pospi committed:
comment UUID helpers lib, remove unneeded method, comment method to deprecate later

pospi committed:
rename crate to more appropriate naming

pospi committed:
pin GraphQL client module to same version as schema module

Connoropolous commented:
I still haven't sent Pospi the cachic secret, that we can add as a secret to the repository, so I will do that now.

Connoropolous commented:
I also just approved the CI runs for this PR, since Harlan and this fork are 'new contributors'

Connoropolous commented:
myself I've never yet tried running the test suite, so I can't comment like Pospi can

Connoropolous commented on .github/workflows/test.yml:
```diff
@@ -0,0 +1,68 @@
+            .cargo/registry/cache/
+            .cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
```
this repo has the rather anomalous behavior of not checking the `Cargo.lock` file into git, which as I understand it has its reasons, but from my experience has had some real downsides

Connoropolous replied:
in my other one, I use `Cargo.toml`, while realizing that that's not best practice

harlantwood replied:
> this repo has the rather anomalous behavior of not checking the `Cargo.lock` file into git, which as I understand it has its reasons, but from my experience has had some real downsides

Hmm I think this is actually best practice for libraries (but not binaries) — https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries

harlantwood replied:
> in my other one, I use `Cargo.toml`, while realizing that that's not best practice

Thanks, should update

Connoropolous replied:
That kinda makes sense. The zomes themselves are more like binaries, while the things in `lib` are libraries, so 🤷 I'm not sure

harlantwood replied:
> That kinda makes sense. The zomes themselves are more like binaries, while the things in `lib` are libraries, so 🤷 I'm not sure

More nuanced suggestion here: https://stackoverflow.com/a/63727523

harlantwood commented:
@pospi have you considered moving to sweetest instead of upgrading
tryorama?  It’s pretty great to have your tests of rust lib in rust.  It’s
slow (unless you’re on an M1 mac) but worth it in my experience so far.

harlantwood commented:
> Tests should be run for any change on sprout, but builds only get run if you push a version tag. 

That’s exactly the behavior of this PR was merged.  And they would run in parallel if you pushed a tag. 

> Eventually though yes I think it's best if everything gets run in GH actions and the build doesn't get kicked off if the tests fail!

Sure, we could chain the builds conditionally later.

harlantwood commented:
> I still haven't sent Pospi the cachic secret, that we can add as a secret to the repository, so I will do that now.

Ah cool.  Still the run was about 5 min, must have been found in the holochain-ci cache.

harlantwood commented:
@pospi  overall I’d suggest you just leave this open until you’re ready to run tests in CI, happy to help more at that time if desired.

pospi committed:
fix release tag match, not a real regex

pospi committed:
remove unnecessary double-handling of null connection URI

Connoropolous commented:
"have you considered moving to sweetest instead of upgrading
tryorama?  It’s pretty great to have your tests of rust lib in rust.  It’s
slow (unless you’re on an M1 mac) but worth it in my experience so far."
I am speaking out of turn, but I do know that the tests actually test the graphql part of the stack as well, so it's not useful to switch off of tryorama

Connoropolous commented on .github/workflows/test.yml:
```diff
@@ -0,0 +1,68 @@
+        if: ${{ runner.os == 'macOS' }}
+        run: sw_vers
+
+      - name: Set up nix
+        uses: cachix/install-nix-action@v16
+        with:
+          nix_path: nixpkgs=channel:nixos-21.05
+          extra_nix_config: |
+            substituters = https://cache.nixos.org https://cache.holo.host https://ci-builds.cachix.org https://holochain-ci.cachix.org
+            trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= cache.holo.host-1:lNXIXtJgS9Iuw4Cu6X0HINLu9sTfcjEntnrgwMQIMcE= cache.holo.host-2:ZJCkX3AUYZ8soxTLfTb60g+F3MkWD7hkH9y8CgqwhDQ= ci-builds.cachix.org-1:fxB0+h/MMlCpXf6hFsQM31YpHbaQoRmcNPNHwDUkXA4= holochain-ci.cachix.org-1:5IUSkZc0aoRS53rfkvH9Kid40NpyjwCMCzwRTXy+QN8=
```
we should consolidate between this way of defining these, and the `nix.conf` way that is currently at play. I have zero attachment to which way it goes

Connoropolous replied:
I am referring to this file, and the command in the Release CI workflow that references it
https://github.com/holo-rea/holo-rea/blob/sprout/.github/nix.conf

pospi committed:
refactor autoconnect logic out of explorer app into Holochain connector module

remove deprecated type alias for DNAMappings

pospi committed:
remove todo

pospi committed:
update scripts to use pnpm commands so workspace is aware of pnpm-specific module IDs (workspace:* etc)

pospi committed:
update CI scripts to use pnpm & remove old sim2h network server startup commands

pospi committed:
pnpm installed locally in GH CI

pospi committed:
unignore Cargo.lock to see if it makes GH actions use same deps

pospi committed:
add Cargo lockfile

Connoropolous committed:
fixes #199 see the issue context
there is a new option for mapZomeFn to instruct it to optionally skip the special encode decode steps

Connoropolous committed:
remove console.log

pospi committed:
Merge pull request #201 from holo-rea/fix-for-199

fix broken agent resolver

pospi committed:
correctly parse datetime format without fractional seconds if returned from backend

Connoropolous committed:
fix the missing entry def for units

Connoropolous committed:
fixes #206

pospi committed:
Merge branch 'sprout' of github.com:holo-rea/holo-rea into sprout

Connoropolous committed:
fixes #208 - update multiple times

the original code didn't suit this context, and it was adapted from h-be/Acorn or hdk_crud.
It didn't suit because in Acorn HeaderHash is used as "id" where as in holo-rea EntryHash is used as "id"
hdk_crud creates a flat update tree, where all updates reference the original header
holo-rea is supposed to create a branching Update tree, which this now allows for.

However, this implementation is still very naive, and assumes that in fact the Update tree
mostly just nests children in one sequence, more like an array than a tree. This won't hold
true in practice, as we're dealing in offline-friendly distributed systems. Solving this properly
relates to issue #196 conflict resolution

Connoropolous committed:
Update record_helpers.rs

pospi committed:
Merge pull request #209 from holo-rea/fix-update-tree

fixes #208 - update multiple times

pospi committed:
Merge pull request #207 from holo-rea/fix-unit-measure

Fix `hasUnit` and `createUnit` bugs

pospi commented:
> @pospi have you considered moving to sweetest instead of upgrading tryorama? It’s pretty great to have your tests of rust lib in rust. It’s slow (unless you’re on an M1 mac) but worth it in my experience so far.

I have, but the decision not to is based on the layer I'm intentionally testing at. There are still tests that use the raw `$cell.call()` methods, but I consider that pattern deprecated and in need of a refactor- ideally all tests should be running through the GraphQL API, because this means the testing is being done against the domain model rather than the Holochain API interface; and in the way that GraphQL binds schemas to the backend you get automatic checking that your "frontend" & "backend" schemas are aligned.

There are some other nice benefits to this in that the test become "how-tos" for prospective developers (the GraphQL is readily discernable among the JavaScript test boilerplate); and the tests can be used against other systems implementing the [ValueFlows GraphQL spec](https://lab.allmende.io/valueflows/vf-schemas/vf-graphql).

pospi commented:
There are also a few extra bits of serialization logic and module binding to other Holochain components in the GraphQL layer and so for hREA UI apps it's best to consider that as the outer boundary of the system. If you're building Rust apps that interact with Holochain directly, then you're looking at more direct interaction with those components anyway and understanding / binding directly to those APIs. In that case, the code in the GraphQL layer for modules [such as agent](https://github.com/holo-rea/holo-rea/blob/sprout/modules/vf-graphql-holochain/queries/agent.ts) might be useful in replicating the logic for adapting standard Holochain concepts & modules to the REA vocabulary.

So far, [agent registration](https://github.com/holochain-open-dev/agent-registration/) and [DNA auth resolver](https://github.com/holochain-open-dev/dna-auth-resolver/) are the only two zome libraries hREA is directly dependent upon, and the way they're included & interact with Rust's crate system is worth exploring in the context of above as I think there could be answers to some rough edges.

> That kinda makes sense. The zomes themselves are more like binaries, while the things in `lib` are libraries, so shrug I'm not sure

Nor am I. The canonical way for reusing code in hApps is something I haven't revised in a while, but there might be a solution if integrated with GH release builds.

- **question:** can we publish zome WASMs to the release pages? (don't see why not...)
- **question:** can we use `url` to configure DNA bundles in place of `bundled`?

This would solve a few issues, expressed by the non-rea zome libraries in `zomes/`. The best practise for including other zome code in an app at the time was to re-export the original crate, [like this](https://github.com/holo-rea/holo-rea/blob/sprout/zomes/dna_auth_resolver/src/lib.rs). The issue becomes unnecessary [duplication / derivation of the original crate name](https://github.com/holo-rea/holo-rea/blob/sprout/zomes/dna_auth_resolver/Cargo.toml#L2), resulting in a different zome hash, losing the ability to provably show that zomes being imported into hREA are unmodified standard modules. My thinking is we could use `url` refs to point to pre-baked WASMs and obviate the need to create these intermediary crates (and make room for updating that process to a Holochain-native deployment method via [DevHub](https://github.com/holochain/devhub-dnas/) at some point).

pospi commented:
The other unanswered question I have on the topic of authoring crates to behave like libraries and binaries is how to do this in a way that Cargo doesn't end up getting confused about.

Currently, the crates compromising the core hREA framework are pinned loosely (`hdk = "0.0"`); and only the zome crates (binaries with `crate_type = "cdylib"`) are pinned to specific HDK versions. The thinking was that this would allow others to import the framework without having to manage niggling dependencies and ask upstream crate authors to bump their HDK dependencies in time with yours, which was what was happening previously and somewhat drained the energy from complex projects in administrative time. So if we can solve some of this above stuff for zome library linking we can remove the last remaining parts of this version coupling issue and DNA authors will be able to mix WASMs compiled for different versions of the HDK without issue so long as those crates are compiled to compatible versions of HDK (which they should be, once HDK moves out of alpha and into proper semver).

Something I've not yet resolved with this is how to correctly set my dependencies across crates to *guarantee* that Cargo will not check out duplicate versions of the HDK and start to create build errors out of duplicate / mismatched versions of the same struct definitions.

Adding `Cargo.lock` to this repo seemed to be a good idea, as it prevented the Github release scripts from getting confused about the correct versions of `hdk` to use with building different parts of the codebase. I *think* what is happening now is that builds run from the hREA project dir use its lockfile as priority and this triggers Cargo to use the versions mentioned there as a priority for other crates in its build chain, even if those crates can potentially reference more recent versions of the same crates. But I need to confirm that somehow. It could just be coincidence.

I also don't know what to expect for consumers of hREA building higher-order Holochain-based functionality on top of it. I have still not committed `Cargo.lock` files to either *DNA Auth Resolver* or *Agent Registration* projects; and these modules are *also* intended to be used as combinations of binaries and libraries in other projects, so it seems logical that they should be following the same crate configuration patterns as hREA (and currently, the absence of lockfiles means they are not).

Not quite sure what my question is in trying to explain this, I guess I'm just trying to solicit feedback on any places where this build and deployment setup can be improved so that there's less friction in working compatible code across an ecosystem.

pospi commented:
Ah

> Users dependent on the library will not inspect the library’s Cargo.lock (even if it exists).

So yes, should put library files in those other modules

pospi committed:
update deps for latest  HDK

pospi committed:
update lockfile

pospi committed:
update nix env for latest Holochain

pospi committed:
update hdk revs in (outdated) module templates

pospi committed:
update for SignedHeaderHashed API change

pospi committed:
fix Unit fields not being public

pospi committed:
update existing entry field validation to use new validations API

pospi committed:
fix method binding for all_agents query, though it doesn't yet seem to be working anyway

pospi committed:
add todo

pospi committed:
fix EconomicResource mising name field

pospi committed:
update clean commands for latest Holonix

pospi committed:
make test package private to prevent workspace-wide publish from including it

pospi committed:
fix PNPM package refs to interpret build folders correctly

pospi committed:
remove build folders for node modules when cleaning

pospi committed:
update lockfile, client now refs unbuilt package but *hopefully* PNPM will resolve this

pospi committed:
add publishing instructions for node packages to developer docs

pospi committed:
mark node packages as non-private, publishConfig.directory handles the src/dist separation now

pospi committed:
update publishConfigs and remove manual package file munging which PNPM should handle itself

pospi committed:
swap out workspace protocol ref in built version of GraphQL client module for actual version of module

this seems dumb. Why doesn't PNPM do this for us?

pospi committed:
update readme for GraphQL client module

pospi committed:
downgrade GraphQL in tests, seems to cause errors with easygraphql-tester not understanding schema types

pospi committed:
fix casing of parameter names in EconomicResource tests

adaburrows committed:
Add pnpm to nix-shell environment.

pospi committed:
update lockfile

pospi committed:
Merge pull request #217 from adaburrows/fix-default-nix

Add pnpm to nix-shell environment.

pospi committed:
update HDK

pospi committed:
update lockfile

pospi committed:
fix link deletion to only target entries in question

pospi committed:
implement alternative more flexible architecture for managing index updates in response to record changes

indexing now supports mixed DnaHashes in identifier data for many:many relationships between Holochain cells

pospi committed:
fix #212 contains & containedIn resolvers for EconomicResource

pospi committed:
export serializeId from test helpers for use in tests which mix low-level Cell API with GraphQL queries

pospi committed:
fix serialization errors in resource_links test; now passing

pospi committed:
return correct paginated Relay structure for ResourceSpecification.conformingResources

pospi committed:
quick first pass at converting index zome return data to Relay-compatible pagination result

pospi committed:
update lockfile

pospi committed:
update conformingResources test for Relay API

pospi committed:
only update EconomicResource container refs if changed

pospi committed:
refix economicResource link resolvers per new indexing API return format

pospi committed:
Merge branch 'sprout' into feature/197-configurable-indexing

pospi committed:
remove old indexing implementation, entirely superceded by automatic indexes

pospi committed:
update all index handler methods to transport-agnostic calling format

pospi committed:
fix tests for EconomicEvent / EconomicResource list API

pospi committed:
start porting test for new Tryorama, but can't get past install anyway

pospi committed:
fix EconomicResource index zome failing EntryDef checks

pospi committed:
presume PNPM to be installed in Github CI as per new Nix config addition

pospi committed:
bump package versions

hackmd-deploy committed:
initial push

Connoropolous committed:
Merge pull request #227 from holo-rea/completions

add a completions.md file which syncs with hackmd

Connoropolous committed:
rename happs folder to dna_bundles

pospi committed:
fix missing Agreement module in Holochain Cell mappings, make all DNAs optional

originally as part of #224

pospi committed:
reorg bundle folders into single toplevel directory

pospi committed:
update readmes for new paths

pospi committed:
update ignore file

pospi committed:
update paths in all configs, helper scripts and test harness

pospi committed:
clean up build clean command

pospi committed:
Merge pull request #228 from holo-rea/rename-happs-to-dna-bundles

fix #225: reorganise bundle file directories

pospi committed:
move DNA bundle files in prep for templating

pospi committed:
move app bundle template, use single template & delete old dev-specific config

pospi committed:
change HC app ID to full suite now the sandbox can support it

pospi committed:
update manifest files & build scripts to allow split dev/release config using same template files

pospi committed:
update Github release script to use new command

pospi committed:
update ignore file

pospi committed:
update docs to reflect new bundle manifest file structure & commands

pospi committed:
move completions file for tracking MVP features to docs dir

Connoropolous committed:
add the VfModule enum, use it throughout, and export it

hackmd-deploy committed:
switched order

Connoropolous committed:
Merge branch 'sprout' into completions

Connoropolous committed:
Merge pull request #237 from holo-rea/completions

completions documentation, order of content change

pospi committed:
add temporary workaround for Tryorama timeout so integration tests can pass

pospi committed:
remove comment

pospi committed:
remove util schema from modules list, it always gets included automatically

pospi committed:
alphabetise default modules for clarity

pospi committed:
Merge pull request #236 from h-REA/234-vf-module-as-enum

add the VfModule enum, use it throughout, and export it

pospi committed:
define VFModule enums as strings to avoid breakage when passed to vf-graphql-holochain (which is a pure-js module)

Connoropolous committed:
make sed command work cross platform mac and linux

Connoropolous committed:
Merge pull request #241 from h-REA/make-cross-platform

make sed command work cross platform mac and linux

Connoropolous committed:
fix basic error blocking test running

Connoropolous committed:
Merge pull request #243 from h-REA/basic-blocking-test-running-fixes

Fix #242: fix basic error blocking test running

Connoropolous committed:
make the integration tests more usable, and more informative

Connoropolous committed:
run them all

Connoropolous committed:
fix economic-resource broken test

Connoropolous committed:
update test report

Commit:
hot fix the recursion bug in get_latest_header_hash

Connoropolous committed:
Merge branch 'sprout' into integration-test-usability

pospi committed:
update agent tests init logic for new tryorama

Commit:
Update resource_logic.js

Commit:
Update package.json

Commit:
Delete integration-tests.sh

Commit:
Update package.json

pospi committed:
bump package versions

pospi committed:
update lockfile

pospi committed:
Merge pull request #245 from h-REA/integration-test-usability

usability goals for integration tests PR

Connoropolous committed:
fix test running so that we don't hit non-deterministic cell order

Connoropolous committed:
add note to dev docs about RUST_LOG tuning

Connoropolous committed:
add a link to further external help

Connoropolous committed:
cleanup

Connoropolous committed:
get test/satisfaction to place where its failing on implementation logic

Commit:
Merge pull request #259 from h-REA/fix-test-running

fix test running so that we don't hit non-deterministic cell order

Connoropolous committed:
add comment in the file about the status of the tests in the file

Connoropolous committed:
extend comment

Commit:
Merge pull request #260 from h-REA/255-test-satisfaction

#255 test satisfaction rust compilation and Runtime error fixes

Connoropolous committed:
skip agent and social-architectures tests for now

Connoropolous committed:
fix the initial error by using mockIdentifier()

Commit:
Merge pull request #261 from h-REA/254-test-flows

#254 test/flows fix the initial errors by using mockIdentifier()

pospi committed:
reduce log verbosity in tests

via #259

pospi committed:
fix config and declarations to correctly wire up Agreement DNA with EconomicEvent & Commitment

pospi committed:
fix config bugs in observation <> planning inter-DNA calls

pospi committed:
fix missing index zome configs for Proposal DNA

pospi committed:
fix more mis-wired indexes in default module arrangement

pospi committed:
fix process index zome names in config

Connoropolous committed:
fix for revisionId which was not getting serialized/deserialized by the mapZomeCall

Connoropolous committed:
addresses #253 -> test_agreement_links.js needs updating to new tryorama (easy)

Connoropolous committed:
fixes for #253, partial progress

Connoropolous committed:
fixes for #246 test/process

Commit:
add backlink to issue tracker

Connoropolous committed:
further fixes for test/fulfillment

Connoropolous committed:
Merge branch 'sprout' of github.com:holo-rea/holo-rea into sprout

pospi committed:
fix agreement index zome missing entrydefs

pospi committed:
sort IDs clientside in agreement link test pending resolution to #220

Connoropolous committed:
fix most of test/specification #249

pospi committed:
fix agreement index update method names in DNA config

pospi committed:
index config fixes for observation, planning & specification DNAs

pospi committed:
fix empty config for locally-hosted index zome throwing an unnecessary error when remote indexes are being updated

pospi committed:
Merge branch 'sprout' of github.com:holo-rea/holo-rea into sprout

Connoropolous committed:
fixes #246 test/process

Connoropolous committed:
almost complete #254 test/flows

Connoropolous committed:
fixes related to #248 test/core-architecture. see that issue for remaining bugs

Connoropolous committed:
fixes #248

pospi committed:
fix nonstandard name for EconomicEvent index zome in Fulfillment

pospi committed:
only check record ID portion of EconomicEvent.fulfills in assertion

temporary solution until split record handling is addressed in #266

pospi committed:
add helpers for sorting lists of records with IDs as raw buffers (for tests not using GraphQL API)

pospi committed:
fix assertions in Fulfillment tests for ordering, list output structs & cross-DNA IDs

pospi committed:
fix satisfaction test assertions with workarounds for sorting, list API changes & multi-DNA awareness

pospi committed:
minor update to assertion message

pospi committed:
first pass at porting Proposal tests to new Tryorama #251

pospi committed:
fix Proposal & Process structs using old Timestamp instead of new DateTime for time fields

pospi committed:
fix missing index config for Proposal zome

pospi committed:
update Proposal CRUD tests to use revisions #251

pospi committed:
fix missing EntryDefs in proposal index zomes

pospi committed:
update ProposedIntent tests to use revisions for update/delete

pospi committed:
fix remote_auth zome not being included in planning DNA

pospi committed:
fix ProposedIntent not declaring EntryDef needed to store credentials for updating planning DNA

pospi committed:
fix missing entryDef in intent index zome

pospi committed:
fix ProposedIntent tests erroring due to undefined ordering

pospi committed:
fix incorrect index name for Proposal.publishedTo

pospi committed:
update ProposedTo tests to use revisions for deletion

pospi committed:
implement ID translation method to allow cross-DNA resolvers to function correctly

pospi committed:
fix Satisfaction & Fulfillment resolvers not dealing with cross-DNA links properly

pospi committed:
export cell remapping helper temporarily for use in tests

pospi committed:
fix assertions for EconomicEvent.satisfies / fulfills by remapping CellIds

pospi committed:
fix passing param adjusted for CellId to satisfaction resolver

pospi committed:
fix errors due to nondeterministic ordering in flow records tests

pospi committed:
workaround result sorting in test assertions for proposedIntent #251

pospi committed:
#254 fix satisfaction result handling helper to match paginated response format

pospi committed:
update logic for anchored records to encode string-based index paths into LinkTag

fixes Unit CRUD tests in #249 by allowing string IDs to be queried from the DHT again, after PathEntry changes made this impossible using Path construct

pospi committed:
#249 fix unit ID assertion in tests to factor in DnaHash portion of ID

pospi committed:
fix wrong type used to constrain Unit deletion logic #249

pospi committed:
remove debug

pospi committed:
remove overly verbose logging in index management methods

pospi committed:
WIP intermediate state for #264 so as to not cause complex config issues.

All index updates & RPC calls now uniformly treated as non-critical and logged for output to assist in debugging.

pospi committed:
remove unneeded storage consts as per #263

pospi commented:
Just noting here that all integration tests are passing again, so we can probably go ahead with this.

I'd also like to see checks configured so that feature branches can't be merged into `sprout` without the integration test suite passing :slightly_smiling_face:

Connoropolous committed:
Update test.yml

Connoropolous committed:
Update test.yml

Connoropolous committed:
Update test.yml

Connoropolous committed:
increase timeout

Connoropolous committed:
Delete config.yml

Connoropolous committed:
Delete .travis.yml

Connoropolous committed:
merge

Connoropolous commented:
closing in favor of #267 since the git histories became diverged in a weird way I had trouble with
